phoenix-core/src/main/java/org/apache/phoenix/coprocessor/GlobalIndexRegionScanner.java

```diff
@@ -82,16 +94,30 @@
     protected long maxLookBackInMills;
     protected IndexToolVerificationResult verificationResult;
     protected IndexVerificationResultRepository verificationResultRepository;
+    protected UngroupedAggregateRegionObserver ungroupedAggregateRegionObserver;
```

Comment:
This superclass does not appear to use the ungroupedAggregateRegionObserver pointer, so it should belong to appropriate subclasses. (Also, we shouldn't be passing coprocs around -- see comments below.)

Reply:
It is using it

Reply:
Could you please point out where? I just expanded the file and I don't see it.

Reply:
You are right that it is not used anymore here in this abstract class but both the classes that expand this class access this attribute. So it is a common attribute.

Reply:
The attribute ungroupedAggregateRegionObserver has been removed from this class